dotCMS/src/main/java/com/dotcms/security/secret/SecretsStoreKeyStoreImpl.java

```diff
@@ -0,0 +1,348 @@
+     * 
+     * @param variableKey
+     * @return
+     * @throws Exception
+     */
+    public Optional<char[]> getValue(final String variableKey) {
+
+        final char[] fromCache = getFromCache(variableKey,
+                () -> loadValueFromStore(variableKey));
+
+        return Arrays.equals(fromCache, CACHE_404) ? Optional.empty() : Optional.ofNullable(decrypt(fromCache));
+
+    }
+
+    /**
+     * tries to load the value from the store on disk
+     * 
+     * @param variableKey
+     * @return
+     */
+    private char[] loadValueFromStore(final String variableKey) {
+        try {
+           final KeyStore keyStore = getSecretsStore();
+            if (keyStore.containsAlias(variableKey)) {
+                final PasswordProtection keyStorePP = new PasswordProtection(loadStorePassword());
+                final SecretKeyFactory factory = SecretKeyFactory.getInstance(SECRETS_STORE_SECRET_KEY_FACTORY_TYPE);
+                final SecretKeyEntry secretKeyEntry = (SecretKeyEntry) keyStore.getEntry(variableKey, keyStorePP);
+                return ((PBEKeySpec) factory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class)).getPassword();
+            } else {
+                return CACHE_404;
+            }
+
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+
+
+    @Override
+    public Set<String> listKeys() {
+        return Sneaky.sneak(this::getKeysFromCache);
+    }
+
+    @Override
+    public synchronized boolean deleteAll(){
+        final File secretStoreFile = new File(secretsKeyStorePath);
+        secretStoreFile.delete();
+        flushCache();
+        return true;
+
+    }
+
+
+    /**
+     * This method saves a secret value into the keystore. The value to be stored is encrypted using the
+     * company key and is used as a "password" for a keystore entry - using it as a passowrd allows us
+     * to store an arbitrary string in a keystore
+     * 
+     * @param variableKey
+     * @param variableValue
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeySpecException
+     * @throws KeyStoreException
+     * @throws Exception
+     */
+    @Override
+    public boolean saveValue(final String variableKey, final char[] variableValue) {
+        try {
+            final SecretKeyFactory factory = SecretKeyFactory.getInstance(SECRETS_STORE_SECRET_KEY_FACTORY_TYPE);
+            final KeyStore keyStore = getSecretsStore();
+            final char [] encryptedVal = encrypt(variableValue).toCharArray();
+            final SecretKey generatedSecret = factory.generateSecret(new PBEKeySpec(encryptedVal));
+            final PasswordProtection keyStorePP = new PasswordProtection(loadStorePassword());
+            keyStore.setEntry(variableKey, new KeyStore.SecretKeyEntry(generatedSecret), keyStorePP);
+            saveSecretsStore(keyStore);
+            putInCache(variableKey, encryptedVal);
+        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)